pkg/resourceapply/apps.go

```diff
@@ -0,0 +1,89 @@
+package resourceapply
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/scylladb/scylla-operator/pkg/naming"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	appv1client "k8s.io/client-go/kubernetes/typed/apps/v1"
+	appv1listers "k8s.io/client-go/listers/apps/v1"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog/v2"
+)
+
+// TODO: write unit test to verify it uses the original RV, if present,
```

Comment:
Please add these tests, these are very important functions. Previous implementation contained tests for PDB and Secrets, this one doesn't have any tests for resources.

Reply:
ok, I've added one for the ApplyStatefulSet, after you get a change to review, I can duplicate it for the other apply functions

Reply:
LGTM, you can copy